src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxStorage.cs

```diff
@@ -31,7 +30,7 @@ public Task Store(OutboxMessage message, OutboxTransaction transaction, ContextB
             var tx = (InMemoryOutboxTransaction) transaction;
             tx.Enlist(() =>
             {
-                if (!storage.TryAdd(message.MessageId, new StoredMessage(message.MessageId, message.TransportOperations.ToList())))
```

Comment:
I thought it is needed because we store the data as any persistence would
do which is essentially copying the data

On Wednesday, July 6, 2016, Brandon Ording <notifications@github.com> wrote:

> In
> src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxStorage.cs
> <https://github.com/Particular/NServiceBus/pull/3903#discussion_r69657132>
> :
>
> > @@ -31,7 +30,7 @@ public Task Store(OutboxMessage message, OutboxTransaction transaction, ContextB
> >              var tx = (InMemoryOutboxTransaction) transaction;
> >              tx.Enlist(() =>
> >              {
> > -                if (!storage.TryAdd(message.MessageId, new StoredMessage(message.MessageId, message.TransportOperations.ToList())))
>
> Should be able to remove this ToArray() call since TransportOperations is
> already an array now, or is there some reason you need a copy of the
> collection?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3903/files/9b3324353ab994bdf2836ce71186e54aed8cdac4#r69657132>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAKosjIqJc-SNMxRh7TnDencXGal1PTdks5qSu08gaJpZM4JFK20>
> .
>